src/Loader.php

```diff
@@ -97,6 +97,25 @@ public function __construct() {
 
 		// Combine JSON translation files (from chunks) when language packs are updated.
 		add_action( 'upgrader_process_complete', array( __CLASS__, 'combine_translation_chunk_files' ), 10, 2 );
+
+		// Handler for WooCommerce and WooCommerce Admin plugin activation.
+		add_action( 'woocommerce_activated_plugin', array( __CLASS__, 'activated_plugin' ) );
+		add_action( 'activated_plugin', array( __CLASS__, 'activated_plugin' ) );
+	}
+
+	/**
+	 * Run when plugin is activated (can be WooCommerce or WooCommerce Admin).
+	 *
+	 * @param string $filename Activated plugin filename.
+	 */
+	public static function activated_plugin( $filename ) {
+		$plugin_domain           = explode( '/', plugin_basename( __FILE__ ) )[0];
+		$activated_plugin_domain = explode( '/', $filename )[0];
+
+		// Ensure we're only running only on activation hook that originates from our plugin.
+		if ( $plugin_domain === $activated_plugin_domain ) {
+			self::generate_translation_strings();
```

Comment:
In this function would it also be worth to check if the files are already generated or not?
Given a user could now first activate WooCommerce, which generates the translations, and then activate WooCommerce admin separately, which would re-generate the translations again.

We can force update on `upgrader_process_complete`.

Reply:
Actually, WooCommerce and WooCommerce Admin use different combined translations if both are installed. This is because the text domain `woocommerce` and `woocommerce-admin` is different for each, thus it's actually necessary to have both translations generated.

As an added point, I feel it's also better to be able to regenerate combined translations at will, for the following reasons:
1. It only runs on plugin activation, thus no perf overhead issues
2. In rare cases where the JSON file might get corrupted, a simple plugin reinstall would solve the problem

Reply:
Thanks, that sounds good to me.